netebpfext/net_ebpf_ext_hook_provider.c

```diff
@@ -743,8 +771,7 @@ net_ebpf_extension_hook_provider_register(
     memset(local_provider_context, 0, sizeof(net_ebpf_extension_hook_provider_t));
     ExInitializePushLock(&local_provider_context->lock);
     InitializeListHead(&local_provider_context->filter_context_list);
-    ExInitializePushLock(&local_provider_context->zombie_list_lock);
-    InitializeListHead(&local_provider_context->zombie_filter_context_list);
+    _ebpf_ext_init_hook_rundown(&local_provider_context->rundown);
```

Comment:
On my testing, this looked to be the root cause of the issues - sometimes, the NMR register was immediately triggering the attach callback, which (without this change) was causing us to use an uninitialized rundown struct.

This fix resolves this issue by moving initialization to before NMR register, ensuring that only valid access is guaranteed.